store.go

```
@@ -118,3 +177,130 @@ func (b *BucketReaderAt) ReadAt(p []byte, off int64) (n int, err error) {
+				rightValue, err = pqarrow.ArrowScalarToParquetValue(e.Value)
+				return false
+			}
+			return true
+		}))
+		if err != nil {
+			return 0, err
+		}
+
+		return rightValue.Uint64(), nil
+	}
+
+	switch e := expr.(type) {
+	case *logicalplan.BinaryExpr:
+		switch e.Op {
+		case logicalplan.OpNotEq:
+			v, err := parseExpr(e)
+			if err != nil {
+				return false, err
+			}
+
+			return v < start && v > end, nil
+		case logicalplan.OpLtEq:
+			v, err := parseExpr(e)
+			if err != nil {
+				return false, err
+			}
+
+			return start <= v, nil
+		case logicalplan.OpLt:
+			v, err := parseExpr(e)
+			if err != nil {
+				return false, err
+			}
+
+			return start < v, nil
+		case logicalplan.OpGt:
+			v, err := parseExpr(e)
+			if err != nil {
+				return false, err
+			}
+
+			return end > v, nil
+		case logicalplan.OpGtEq:
+			v, err := parseExpr(e)
+			if err != nil {
+				return false, err
+			}
+
+			return end >= v, nil
+		case logicalplan.OpEq:
+			v, err := parseExpr(e)
+			if err != nil {
+				return false, err
+			}
+
+			return v >= start && v <= end, nil
+		case logicalplan.OpAnd:
+			left, lefterr := compareTimestamp(col, start, end, e.Left)
+			if lefterr != nil && lefterr != ErrNonTimestampCol {
+				return false, lefterr
+			}
+
+			right, righterr := compareTimestamp(col, start, end, e.Right)
+			if righterr != nil && righterr != ErrNonTimestampCol {
+				return false, righterr
+			}
+
+			return (left || lefterr != nil) && (right || righterr != nil), nil
+		case logicalplan.OpOr:
+			left, lefterr := compareTimestamp(col, start, end, e.Left)
+			if lefterr != nil && lefterr != ErrNonTimestampCol {
+				return false, lefterr
+			}
+
+			right, righterr := compareTimestamp(col, start, end, e.Right)
+			if righterr != nil && righterr != ErrNonTimestampCol {
+				return false, righterr
+			}
+
+			return left || right, nil
+		default:
+			// unsupported op; return true. Better to have false positives
+			return true, nil
```

If possible we should at least log this. This actually bit us before, where we silently ignore things and it was super hard to find in the end...

---

I actually don't think we should do anything here. This is a safe fallback, where the worst thing that can happen is that we look at the block. The only thing that the place where this bit us had in common was a type switch, but that's where the similarity ends in my opinion. I don't see a need for anything other than tests for the cases that we want to ensure work, which this has.